rewrite-xml/src/main/java/org/openrewrite/xml/ChangeNamespaceValue.java

```diff
@@ -0,0 +1,160 @@
+            example = "property",
+            required = false)
+    String elementName;
+
+    @Nullable
+    @Option(displayName = "Old value",
+            description = "Only change the property value if it matches the configured `oldValue`.",
+            example = "newfoo.bar.attribute.value.string",
+            required = false)
+    String oldValue;
+
+    @Option(displayName = "New value",
+            description = "The new value to be used for the namespace.",
+            example = "newfoo.bar.attribute.value.string")
+    String newValue;
+
+    @Nullable
+    @Option(displayName = "Resource version",
+            description = "The version of resource to change",
+            example = "1.1",
+            required = false)
+    String versionMatcher;
+
+    @Nullable
+    @Option(displayName = "Search All Namespaces",
+            description = "Specify whether evaluate all namespaces. Defaults to true",
+            example = "true",
+            required = false)
+    Boolean searchAllNamespaces;
+
+
+    @JsonCreator
+    public ChangeNamespaceValue(@Nullable @JsonProperty("elementName") String elementName, @Nullable @JsonProperty("oldValue") String oldValue,
+                                @NonNull @JsonProperty("newValue") String newValue, @Nullable @JsonProperty("versionMatcher") String versionMatcher,
+                                @Nullable @JsonProperty("searchAllNamespaces") Boolean searchAllNamespaces) {
+        this.elementName = elementName;
+        this.oldValue = oldValue;
+        this.newValue = newValue;
+        this.versionMatcher = versionMatcher;
+        this.searchAllNamespaces = searchAllNamespaces;
+    }
+
+    @Override
+    public TreeVisitor<?, ExecutionContext> getVisitor() {
+
+        return new XmlIsoVisitor<ExecutionContext>() {
+
+            @Nullable
+            private final XPathMatcher elementNameMatcher = elementName != null ? new XPathMatcher(elementName) : null;
+
+            @Override
+            public Xml.Tag visitTag(Xml.Tag tag, ExecutionContext ctx) {
+                Xml.Tag t = super.visitTag(tag, ctx);
+
+                if (matchesElementName(getCursor()) && matchesVersion(t)) {
+                    t = t.withAttributes(ListUtils.map(t.getAttributes(), this::maybeReplaceNamespaceAttribute));
+                }
+
+                return t;
+            }
+
+            private boolean matchesElementName(Cursor cursor) {
+                return elementNameMatcher == null || elementNameMatcher.matches(cursor);
+            }
+
+            private boolean matchesVersion(Xml.Tag tag) {
+                if (versionMatcher != null) {
+                    for (Xml.Attribute attribute : tag.getAttributes()) {
+                        if (isVersionAttribute(attribute) && isVersionMatch(attribute)) {
+                            return true;
+                        }
+                    }
+                    return false;
+                }
+                return true;
```

Comment:
```suggestion
                if (versionMatcher == null) {
                    return true;
                }
                for (Xml.Attribute attribute : tag.getAttributes()) {
                    if (isVersionAttribute(attribute) && isVersionMatch(attribute)) {
                        return true;
                    }
                }
                return false;
```

Reply:
Figured have the method read true -> true -> false, instead of true -> false -> true.